kale/embed/mpca.py

```diff
@@ -1,49 +1,76 @@
-# Principal Component Analysis of Tensor Objects", IEEE Transactions on Neural 
-# Networks, Vol. 19, No. 1, Page: 18-39, January 2008.
 # =============================================================================
+
+"""Python implementation of Multilinear Principal Component Analysis (MPCA)
+Includeing implementaion as a scikit-learn object and an independent function.
+
+Reference:
+    Haiping Lu, K.N. Plataniotis, and A.N. Venetsanopoulos, "MPCA: Multilinear
+    Principal Component Analysis of Tensor Objects", IEEE Transactions on Neural 
+    Networks, Vol. 19, No. 1, Page: 18-39, January 2008. For initial Matlab
+    implementation, please go to https://uk.mathworks.com/matlabcentral/fileexchange/26168.
+"""
+
 import numpy as np
-import tensorly as tl
-from tensorly.base import fold, unfold
-from tensorly.tenalg import mode_dot, multi_mode_dot
+# import tensorly as tl
+from tensorly.base import unfold
+from tensorly.tenalg import multi_mode_dot
 from sklearn.base import BaseEstimator, TransformerMixin
 
 
 class MPCA(BaseEstimator, TransformerMixin):
-    def __init__(self, variance_explained=0.97, max_iter=1):
-        """
-        Implementation compatible with sickit-learn
-        ----------
-        Parameter:
-            variance_explained: ration of variance to keep (between 0 and 1)
-            max_iter: max number of iteration, integer
-        ----------
-        Attributes:
-            tPs: list of projection matrices
-        """
-        self.var_exp = variance_explained
+    """MPCA implementation compatible with sickit-learn
+
+    Args:
+        var_explained (float, optional): ration of variance to keep
+            (between 0 and 1). Defaults to 0.97.
+        max_iter (int, optional): max number of iteration. Defaults to 1.
+
+    Attributes:
+        tPs: list of projection matrices
+
+    Examples:
+        >>> import numpy as np
+        >>> from kale.embed.mpca import MPCA
+        >>> X = np.random.random((20, 25, 20, 40))
+        >>> X.shape
+        (20, 25, 20, 40)
+        >>> mpca = MPCA(variance_explained=0.9)
+        >>> X_transformed = mpca.fit_transform(X)
+        >>> X_transformed.shape
+        (18, 23, 18, 40)
+        >>> X_inverse = mpca.inverse_transform(X_transformed)
+        >>> X_inverse.shape
+        (20, 25, 20, 40)
+    """
+    def __init__(self, var_explained=0.97, max_iter=1):
+        self.var_explained = var_explained
         self.max_iter = max_iter
-        # self.Xmean = None
-        # self.n_dim = None
         self.tPs = []
 
-    def fit(self, X):
-        """
-        Parameter:
-            X: array-like, ndarray of shape (I1, I2, ..., n_samples)
-        ----------
-        Return:
-            self
+    def fit(self, X, y=None):
+        """Fit the model with X.
+
+        Args
+            X (ndarray): input data, shape (I1, I2, ..., n_samples), where n_samples
+                is the number of samples, I1, I2 ... are the dimensions of corresponding
```

Comment:
I1, I2, .... does not specify when to end.
Will I_1, ..., I_N or I1, I2, .... be better?

Reply:
Yes, will change

Reply:
Sorry, I mean Will I_1, ..., I_N or I1, I2, ....I_N be better? (missed I_N)